python-mini-projects/projects/Time_to_load_website/time_to_load_website.py

```diff
@@ -0,0 +1,31 @@
+from urllib.request import urlopen
+import time
+
+
+def get_load_time(url):
+    """This function takes a user defined url as input
+    and returns the time taken to load that url in seconds.
+
+    Args:
+        url (string): The user defined url.
+
+    Returns:
+        time_to_load (float): The time taken to load the website in seconds.
+    """
+
+    if ("https" or "http") in url:  # Checking for presence of protocols
+        open_this_url = urlopen(url)  # Open the url as entered by the user
+    else:
+        open_this_url = urlopen("https://" + url)  # Adding https to the url
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *open\_this\_url*. The resource is allocated by call *request.urlopen*. The resource is referenced at lines 21, 23. The resource is closed at line 23. There are other execution paths that do not contain closure statements, for example, when *time()* throws an exception. To prevent this resource leak, close *open\_this\_url* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).